src/base/model.js

```diff
@@ -243,16 +243,29 @@ ModelBase.prototype.isNew = function() {
  * @returns {Object} Serialized model as a plain object.
  */
 ModelBase.prototype.serialize = function(options = {}) {
-  const { shallow = false, omitPivot = false, ...rest } = options;
+  const { shallow = false, omitPivot = false, omitNew = false, ...rest } = options;
   const { attributes } = this;
 
+  if (omitNew && this.isNew()) {
+    return null;
+  }
+
   if (!shallow) {
 
-    const relations = mapValues(this.relations, (relation, key) =>
-      relation.toJSON != null
-        ? relation.toJSON({ shallow, omitPivot, ...rest })
-        : relation
-    );
+    let relations = mapValues(this.relations, (relation, key) => {
+      if (relation.toJSON != null) {
+        let rel = relation.toJSON(options);
+        if (_.isArray(rel)) {
+          // Filter null values in a collection coming from the omitNew option
+          rel = _.filter(rel, x => ! _.isNull(x));
+        }
+        return rel;
+      }
+        relation
```

Comment:
Currently `omitNew` doesn't work for collections.
```js
var collection = new SomeCollection([{id: 1}, {noid: null}, {id: 3}, {noid: null}]);
collection.toJSON({omitNew: true}); // => [{id: 1}, null, {id: 3}, null]
```
This case is not covered by your implementation nor tests.